src/scielo/bin/xml/app_modules/generics/exporter.py

```diff
@@ -3,75 +3,155 @@
                 )
             else:
                 return zip_file_path
 
-    def export_by_ftp(self, local_file_path):
+    def export_by_ftp(self, local_file_path, server, user, password, remote_path):
+        background = AsyncFTP(local_file_path, server, user, password, remote_path)
+        background.start()
+
+
+class AsyncFTP(threading.Thread):
+    def __init__(self, local_file_path, server, user, password, remote_path, timeout=60):
+        threading.Thread.__init__(self)
+        self.local_file_path = local_file_path
+        self.server = server
+        self.user = user
+        self.password = password
+        self.remote_path = remote_path
+        self.timeout = timeout
+
+    def run(self):
+        exp_logger.info("FTP.START")
         try:
-            config = self.ftp_config
-        except KeyError:
-            logging.info("Export: Invalid configuration")
-        else:
-            server, user, password, remote_path = config
+            ftp = FTP(self.server, self.user, self.password, self.timeout)
+        except all_errors as e:
+            exp_logger.info(e)
+            return
+        try:
+            if self.remote_path:
+                ftp.cwd(self.remote_path)
+            exp_logger.info("ftp " + self.local_file_path)
+            remote_name = os.path.basename(self.local_file_path)
+            with open(self.local_file_path, 'rb') as f:
+                try:
+                    exp_logger.info("FTP.STOR %s - start" % remote_name)
+                    ftp.storbinary('STOR {}'.format(remote_name), f)
+                    exp_logger.info("FTP.STOR %s - end" % remote_name)
+                except all_errors:
+                    exp_logger.info(
+                        'FTP: Unable to send %s to %s' %
+                        (self.local_file_path, remote_name), exc_info=True)
+        except all_errors as e:
+            exp_logger.info("all_errors as e")
+
+            exp_logger.info(e)
+        finally:
+            ftp.close()
+            exp_logger.info("FTP.END")
             try:
-                with FTP(server, timeout=60) as ftp:
-                    ftp.login(user, password)
-                    if remote_path:
-                        ftp.cwd(remote_path)
-                    remote_name = os.path.basename(local_file_path)
-                    with open(local_file_path, 'rb') as f:
-                        try:
-                            ftp.storbinary('STOR {}'.format(remote_name), f)
-                        except all_errors:
-                            logging.info(
-                                'FTP: Unable to send %s to %s' %
-                                (local_file_path, remote_name), exc_info=True)
-            except all_errors:
-                logging.info("Unable to transfer: %s" % local_file_path)
+                os.unlink(self.local_file_path)
+                shutil.rmtree(os.path.dirname(self.local_file_path))
```

Comment:
Faz sentido que o arquivo seja removido na origem após a transferência? Eu não esperaria por este efeito colateral se fosse usar esta classe.

Reply:
@gustavofonseca, analisando o código anterior, não apagar este arquivo alteraria o comportamento já existente no exportador atual. Faz sentido neste momento analisar a necessidade da mudança neste comportamento?

Reply:
Dada a baixa probabilidade de olharmos novamente para este trecho de código nos próximos meses, eu penso que tudo que for identificado deve ser apontado. Isso não quer dizer que os pontos deverão ser resolvidor no escopo deste PR, mas ignorá-los não me parece o melhor caminho.

Reply:
@gustavofonseca seria esse o caso para abrimos _tickets_ indicando os pontos a serem melhorados?